src/test/regress/spec/isolation_citus_pause_node.spec

```diff
@@ -0,0 +1,171 @@
+setup
+{
+	SET citus.shard_replication_factor to 1;
+
+    CREATE TABLE company(id int primary key, name text);
+	select create_distributed_table('company', 'id');
+
+	create table employee(id int , name text, company_id int  );
+	alter table employee add constraint employee_pkey primary key (id,company_id);
+
+	select create_distributed_table('employee', 'company_id');
+
+	insert into company values(1,'c1');
+	insert into company values(2,'c2');
+	insert into company values(3,'c3');
+
+	insert into employee values(1,'e1',1);
+	insert into employee values(2,'e2',1);
+	insert into employee values(3,'e3',1);
+
+	insert into employee values(4,'e4',2);
+	insert into employee values(5,'e5',2);
+	insert into employee values(6,'e6',2);
+
+	insert into employee values(7,'e7',3);
+	insert into employee values(8,'e8',3);
+	insert into employee values(9,'e9',3);
+	insert into employee values(10,'e10',3);
+
+
+}
+
+teardown
+{
+    DROP TABLE company,employee;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-node-not-found"
+{
+	DO $$
+	DECLARE
+		v_node_id int := -1;
+		v_node_exists boolean := true;
+		v_count int := -1;
+		v_exception_message text;
+		v_expected_exception_message text := '';
+	BEGIN
+
+		-- Get a node-id that does not exist in the cluster
+
+		while v_node_exists loop
```

Comment:
You can use nextval(<sequence_name>) to get a non-existent node instead of the loop.

Reply:
Fixed